### PR TITLE
Added handler runner class

### PR DESCRIPTION
**Pull request by tomaszrybicki**
Signed-off-by: Tomasz Rybicki <tomasz.rybicki@intel.com>

### CODE REVIEW

File: source/octf/trace/parser/HandlerRunner.h [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+class HandlerRunner {
+    static_assert(std::is_base_of<TraceEventHandler<Event>, Handler>(),
+                  "Attempted to instantiate template with wrong event type.");
+
+public:
+    HandlerRunner() = default;
+    virtual ~HandlerRunner() = default;
+
+    /**
+     * @brief Add handler to runner
```
Comment (mariuszbarczak) [inline source/octf/trace/parser/HandlerRunner.h]: Adds

File: source/octf/trace/parser/HandlerRunner.h [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+     * @param handler Handler to be run
+     */
+    void addHandler(std::shared_ptr<Handler> handler) {
+        m_handlers.push_back(handler);
+    }
+
+    // TODO (trybicki) Limit thread number based on available cpus and/or memory
+    /**
+     * @brief Run all added handlers
```
Comment (mariuszbarczak) [inline source/octf/trace/parser/HandlerRunner.h]: Runs

File: source/octf/trace/parser/HandlerRunner.h [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+        for (auto &job : m_jobs) {
+            job.join();
+        }
+
+        m_finished = true;
+    }
+
+    /**
+     * @brief Get a list of handlers which finished processing
```
Comment (mariuszbarczak) [inline source/octf/trace/parser/HandlerRunner.h]: Gets

File: source/octf/trace/parser/TraceEventHandler.h [hunk 4]
```diff
@@ -97,6 +97,8 @@ class TraceEventHandler {
         return true;
     }
 
+    typedef EventType eventType;
```
Comment (mariuszbarczak) [inline source/octf/trace/parser/TraceEventHandler.h]: Suggestion: HandlerEventType

Comment (tomaszrybicki) [reply source/octf/trace/parser/TraceEventHandler.h]: I prefer to use the shorter version, I think it's still clear what kind of event it is.

**Pull request merged**